tests/useful_mocks.py

```diff
@@ -5,27 +5,48 @@
 from unittest_protocols import Ability, Action, Question
 
 
-def get_mock_action(**kwargs):
+def get_mock_action(**kwargs) -> mock.Mock:
     mock_action = mock.create_autospec(Debug(), _name="fakeaction", instance=True)
     mock_action.perform_as = mock.create_autospec(Debug().perform_as, **kwargs)
     mock_action.describe.return_value = "An African or a European swallow?"
     return mock_action
 
 
-def get_mock_question():
-    return mock.create_autospec(Question, instance=True)
+def get_mock_question() -> mock.Mock:
+    question = mock.create_autospec(Question, instance=True)
+    question.describe.return_value = "Description of Mocked Question"
+    question.answered_by.return_value = True
+    return question
 
 
-def get_mock_resolution():
+def get_mock_resolution() -> mock.Mock:
     return mock.create_autospec(BaseResolution, instance=True)
 
 
-def get_mock_task():
+def get_mock_task() -> mock.Mock:
     """Get a describable mock task."""
     task = mock.create_autospec(Action, instance=True)
     task.describe.return_value = "A mocked task."
     return task
 
 
-def get_mock_ability():
+def get_mock_ability() -> mock.Mock:
     return mock.create_autospec(Ability, instance=True)
+
+
+class FakeAbility:
+    def forget(self):
+        return
+
+
+class AnotherFakeAbility:
+    def forget(self):
+        return
```

Comment:
Given that we have the `get_mock_ability` function above, do we need to have these two `FakeAbility` classes in here? Can't we just do `FakeAbility = get_mock_ability()` two times?

Reply:
I created that in order to feel good about the testing actor (see below example). I'm not confident in creating a mock object that has the same capability.  

```
    actor = Actor.named("Tester").who_can(FakeAbility())
    assert actor.has_ability_to(FakeAbility)
```

Reply:
Summarizing our discussion on Discord, it seems that we have two options for the Ability part of this nut.

We could use `Ability` instead, which looks like this in the tests:
```python
Tester = AnActor.named("Tester").who_can(FakeAbility())
Tester.uses_ability_to(Ability)  # a little strange, where did Ability come from?
```

Or, we could just use our `FakeAbility` class; though i'd still prefer to have a `get_mock_ability_class` function that will create these `FakeAbility` classes on the fly, and maybe inherit from `mock.Mock` so we get those benefits too. Like this:
```python
def get_mock_ability_class():
    
    class FakeAbility(mock.Mock()):
        """Made by Totally Real Abilities Inc."""
        def forget(self):
            return
 
    return FakeAbility
```

This way we don't need to keep making `AnotherAnotherFakeAbility` when we need more and more (if we do).

Reply:
I don't think this will allow the following test to pass, which is why I ended up making two classes. Perhaps there is another way?
```python
def test_has_ability() -> None:
    actor = Actor.named("Tester").who_can(FakeAbility())

    assert actor.has_ability_to(FakeAbility)
    assert not actor.has_ability_to(AnotherFakeAbility)
```

Reply:
After more trial and error and discussion, here's our final answer for posterity:

```python
def get_mock_ability_class():

    class FakeAbility(Ability):
        def __new__(cls, *args, **kwargs):
            return mock.create_autospec(FakeAbility, *args, **kwargs)

    return FakeAbility
```

Reply:
For posterity, this is lovely and magical.   Bravo